booking/src/main/java/org/dcsa/conformance/standards/booking/checks/BookingChecks.java

```diff
@@ -0,0 +1,684 @@
+    (body) -> {
+      var bookingStatus = body.path("bookingStatus");
+      var amendedBookingStatus = body.path("amendedBookingStatus");
+      var issues = new LinkedHashSet<String>();
+      var status = amendedBookingStatus.isMissingNode() || amendedBookingStatus.isNull() ? bookingStatus : amendedBookingStatus;
+      if (REASON_ABSENCE_STATES.contains(BookingState.fromWireName(status.asText()))) {
+        if (body.hasNonNull("reason")) {
+          issues.add("reason must not be in the Booking States %s".formatted(REASON_ABSENCE_STATES));
+        }
+      }
+      return issues;
+    }
+  );
+
+
+  private static final JsonContentCheck CHECK_ABSENCE_OF_CONFIRMED_FIELDS = JsonAttribute.customValidator(
+    "check absence of confirmed fields in non confirmed booking states",
+    body -> {
+      var issues = new LinkedHashSet<String>();
+      var bookingStatus = body.path("bookingStatus").asText("");
+      if (NON_CONFIRMED_BOOKING_STATES.contains(BookingState.fromWireName(bookingStatus))) {
+        if (body.hasNonNull("termsAndConditions")) {
+          issues.add("termsAndConditions must not be present in %s".formatted(bookingStatus));
+        }
+        if (body.hasNonNull("carrierClauses")) {
+          issues.add("carrierClauses must not be present in %s".formatted(bookingStatus));
+        }
+        if (body.hasNonNull("charges")) {
+          issues.add("charges must not be present in %s".formatted(bookingStatus));
+        }
+        if (body.hasNonNull("advanceManifestFilings")) {
+          issues.add("advanceManifestFilings must not be present in %s".formatted(bookingStatus));
+        }
+        if (body.hasNonNull("confirmedEquipments")) {
+          issues.add("confirmedEquipments must not be present in %s".formatted(bookingStatus));
+        }
+        if (body.hasNonNull("transportPlan")) {
+          issues.add("transportPlan must not be present in %s".formatted(bookingStatus));
+        }
+        if (body.hasNonNull("shipmentCutOffTimes")) {
+          issues.add("shipmentCutOffTimes must not be present in %s".formatted(bookingStatus));
+        }
+      }
+      return issues;
+    });
+
+  private static final JsonContentCheck CHECK_CONFIRMED_BOOKING_FIELDS = JsonAttribute.customValidator(
+    "check confirmed booking fields availability",
+    body -> {
+      var issues = new LinkedHashSet<String>();
+      var bookingStatus = body.path("bookingStatus").asText("");
+      if (CONFIRMED_BOOKING_STATES.contains(BookingState.fromWireName(bookingStatus))) {
+        if (body.get("confirmedEquipments") == null) {
+          issues.add("confirmedEquipments for confirmed booking is not present");
+        }
+        if (body.get("transportPlan") == null) {
+          issues.add("transportPlan for confirmed booking is not present");
+        }
+        if (body.get("shipmentCutOffTimes") == null) {
+          issues.add("shipmentCutOffTimes for confirmed booking is not present");
+        }
```

Comment:
I talked with Henrik, and the condition here is that must be non-empty.  I think the conditions can be written as `if (body.path("...").isEmpty())`

The presence check is insufficient as it enables / allows an empty list to be provided.